library/core/src/cell.rs

```diff
@@ -1811,6 +1811,8 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 ///
 /// [`.get_mut()`]: `UnsafeCell::get_mut`
 ///
+/// `UnsafeCell<T>` has the same in-memory representation as its inner type `T`.
+///
```

Comment:
I think this is a good idea.

But I'll also note that your initial PR comment contains a wealth of useful information. What do you think about adapting some or most of it to include in the docs here?

Reply:
@BurntSushi that's a nice idea, I'll update the PR!

Reply:
I just pushed the additional docs :+1: